File: shopping-cart-backend/src/main/java/com/basil/shoppingcart/service/AdminUserService.java
package com.basil.shoppingcart.service;

import java.util.List;

import com.basil.shoppingcart.dto.response.UserResponse;

public interface AdminUserService {

    List<UserResponse> getAllUsers();

    UserResponse getUserById(
            Long userId
    );

    UserResponse activateUser(
            Long userId
    );

    UserResponse deactivateUser(
            Long userId
    );
}

File: shopping-cart-frontend/src/services/adminUserService.js
import apiClient from "./apiClient";

const adminUserService = {
    getAllUsers: async () => {
        const response = await apiClient.get("/admin/users");
        return response.data;
    },

    activateUser: async (userId) => {
        const response = await apiClient.put(`/admin/users/${userId}/activate`);
        return response.data;
    },

    deactivateUser: async (userId) => {
        const response = await apiClient.put(`/admin/users/${userId}/deactivate`);
        return response.data;
    }
};

export default adminUserService;